ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayDimensions/DisplayDimensions.qml

```diff
@@ -0,0 +1,149 @@
+                            mmpk.maps[0].operationalLayers.get(indexOfDimensionLayer).definitionExpression = "DIMLENGTH >= 450";
+                        }
+                        else {
+                            mmpk.maps[0].operationalLayers.get(indexOfDimensionLayer).definitionExpression = "";
+                        }
+                    }
+                    enabled: mmpk.loadStatus === Enums.LoadStatusLoaded && visibilityToggle.checked ? true : false
+                }
+            }
+        }
+
+        // Pop-up error message box
+        MessageDialog {
+            id: errorMessageBox
+            title: "Error message!"
+            text: errorMessage
+            icon: StandardIcon.Warning
+            visible: errorMessage === "" ? false : true;
+            onAccepted: errorMessage = "";
+        }
+    }
+
+    Component.onCompleted: mmpk.load();
+
+    MobileMapPackage {
+        id: mmpk
+        path: dataPath + "Edinburgh_Pylon_Dimensions.mmpk"
+
+        onLoadStatusChanged: {
+            // If an error occurred during loading, pass the error onto the handleError function.
+            if (loadStatus === Enums.LoadStatusFailedToLoad) {
+                handleError(mmpk.loadError);
+            }
+
+            // If the mmpk has not loaded, return.
+            if (loadStatus !== Enums.LoadStatusLoaded) {
+                return;
+            }
+
+            // If there is more than one map, return.
+            if (mmpk.maps.length > 1) {
+                return;
+            }
+
+            // Set viewpoint.
+            const viewpointPoint = ArcGISRuntimeEnvironment.createObject("Point", { x: -3.3098678, y: 55.9074044, spatialReference: Factory.SpatialReference.createWgs84()});
+            const viewpointCenter = ArcGISRuntimeEnvironment.createObject("ViewpointCenter", {center: viewpointPoint, targetScale: 30000});
+            mmpk.maps[0].initialViewpoint = viewpointCenter;
+
+            // Set the map view's map to the first map in the mobile map package
+            mapView.map = mmpk.maps[0];
+            // Set the minimum scale as 1:35000 to prevent zooming out too far.
+            mapView.map.minScale = 35000;
+
+            // Loop through all layers in the mmpk and find the dimension layer.
+            for (let counter = 0; counter < mmpk.maps[0].operationalLayers.count; counter++) {
+                // Check each layer to see if it is a DimensionLayer.
+                if (mmpk.maps[0].operationalLayers.get(counter).layerType === Enums.LayerTypeDimensionLayer) {
+                    // Save the index of the DimensionLayer - it is used to control layer visibility and definition expressions.
+                    indexOfDimensionLayer = counter;
+                    // Use the name of the DimensionLayer to define the title of the UI controls.
+                    toggleBoxTitle.text = mmpk.maps[0].operationalLayers.get(counter).name;
```

Comment:
you need to add a `break` here so you don't keep searching after you find the layer you are looking for